Libs/Visualization/VTK/Widgets/ctkVTKWidgetsUtils.cpp

```diff
@@ -99,33 +99,37 @@ QImage ctk::vtkImageDataToQImage(vtkImageData* imageData)
   int width = imageData->GetDimensions()[0];
   int height = imageData->GetDimensions()[1];
   vtkIdType numberOfScalarComponents = imageData->GetNumberOfScalarComponents();
-  QImage image;
+
+  QImage::Format pixelFormat;
   if (numberOfScalarComponents == 3)
-  {
-    image = QImage(width, height, QImage::Format_RGB888);
-  }
+    {
+    pixelFormat = QImage::Format_RGB888;
+    }
   else if (numberOfScalarComponents == 4)
-  {
-    image = QImage(width, height, QImage::Format_RGBA8888);
-  }
+    {
+    pixelFormat = QImage::Format_RGBA8888;
+    }
 #if QT_VERSION >= QT_VERSION_CHECK(5,5,0)
   else if (numberOfScalarComponents == 1)
-  {
-    image = QImage(width, height, QImage::Format_Grayscale8);
-  }
+    {
+    pixelFormat = QImage::Format_Grayscale8;
+    }
 #endif
   else
-  {
+    {
     // unsupported pixel format
     return QImage();
-  }
+    }
+
+  int bytesPerLine = imageData->GetIncrements()[1]; // line increment
 
-  unsigned char* qtImageBuffer = image.bits();
-  memcpy( qtImageBuffer,
-    imageData->GetPointData()->GetScalars()->GetVoidPointer(0),
-    numberOfScalarComponents * width * height);
+  // Create an image from a buffer. The QImage object does not take ownership of the voxel buffer.
+  QImage image(static_cast<uint8_t*>(imageData->GetPointData()->GetScalars()->GetVoidPointer(0)),
```

Comment:
shouldn't it be a `reinterpret_cast` rather than a `static_cast` ?

Reply:
fixed!